portal-web/docroot/html/taglib/ui/search_iterator/lexicon/list.jsp

```diff
@@ -276,7 +276,7 @@ if (fixedHeader) {
 					}
 				%>
 
-					<tr class="<%= GetterUtil.getString(row.getClassName()) %> <%= row.getCssClass() %> <%= row.getState() %> <%= rowIsChecked ? "active" : StringPool.BLANK %>" data-qa-id="row" <%= AUIUtil.buildData(data) %>>
+					<tr class="<%= GetterUtil.getString(row.getClassName()) %> <%= row.getCssClass() %> <%= row.getState() %> <%= rowIsChecked ? "active" : StringPool.BLANK %>" data-qa-id="row" id="<portlet:namespace />search_row_<%= String.valueOf(i)%>_<%= row.getRowId() %>" <%= AUIUtil.buildData(data) %>>
```

Comment:
I think maybe this could still generate duplicated ids if more than one search container is on the page and especially if `rowPropertyId` is not used because all rows will get their index.

Related to that last part, sounds like `i` might be unnecessary because if there's no `rowPropertyId`, then `i === `row.getRowId()` and if it is, the id should already do enough to uniquely id the element?

Another concern I have with this is that in most cases (no `rowPropertyId`) the `id` might not be really helpful because different elements will get the same id `_com_foo_myPortlet_1_1` will always be first row in any search container (if I'm reading this correctly), so not sure about how valuable this would be for a generic use case. Maybe we can only do this if a non-numeric id is specified via `rowPropertyId`?

Reply:
I guess you mean if more than one search container is on the page and it's being generated by the same portlet. And yes, in that case, they would be duplicated.

The reason why I added the index in my effort to make the ids unique is because otherwise we need to trust rowPropertyId passed to the taglib is going to be unique.

> not sure about how valuable this would be for a generic use case.
Completely agree. That's the reason why in the beginning I hesitated whether consider this as a product limitation.

> Maybe we can only do this if a non-numeric id is specified via rowPropertyId?
Not sure I got this point. If we are passing userId (numeric) as rowPropertyId why not to use it as id HTML attribute (prefixed by portlet's namespace)?

Reply:
> Not sure I got this point. If we are passing userId (numeric) as rowPropertyId why not to use it as id HTML attribute (prefixed by portlet's namespace)?


My bad, I wasn't clear there. I meant a non-null value of `rowPropertyId` as a way to more explicitly signal uniqueness of the key. As you showed, [by default we return `rowIndex` as an id](https://github.com/liferay/liferay-portal/blob/master/util-taglib/src/com/liferay/taglib/ui/SearchContainerRowTag.java#L317-L334) which does not univocally map to a given entity, thus having no real value in identifying it. Hopefully that's more clear :)

Reply:
> Completely agree. That's the reason why in the beginning I hesitated whether consider this as a product limitation.

Do you have a specific use case coming from a customer? Why do they need the `id`? I thought I saw something, but neither the LPS nor the LPP seem to explain it other than "allow `id`", but since you're involved I'm assuming there is at least some customer looking to do something with this.

I would lean towards also mark this as a product limitation or at least try to see if there's a better way to address the use case since this seems to potentially create many sharp edges...

Reply:
What they want to do is to retrieve vía javascript `tr` HTML element based on the property model they are representing. A brief example:
If our Search Container is showing all users in the platform they want to retrieve the row corresponding to some specific userId.

Regarding rowIdProperty is defined in our .tld as:
_A model property to use as the row ID_

They thought passing `rowIdProperty: "userId"` was going to add that id to every row, so later they could get the row doing something like `document.getElementById(userId)` and manipulate it according to their requirements.

Do you have any other idea about how to retrieve a row based on a property model?

Reply:
> They thought passing rowIdProperty: "userId" was going to add that id to every row, so later they could get the row doing something like document.getElementById(userId) and manipulate it according to their requirements.

I think this is a reasonable expectation. It is predicated on the fact that they passed a non-null `rowPropertyId` so I think maybe that should at least be part of the solution.

> What they want to do is to retrieve vía javascript tr HTML element based on the property model they are representing

I assume they are then trying to make some modifications in DOM to existing elements? This can have unexpected results in several instances:
- The element is not really there (pagination, sorting, filtering...)
- The row is not meant to be mutated (because some other component maintains state...)
- Some changes may be destructive (modifying the row structure, its selection input...)

In any case, I think the expectation makes sense as long as we don't open the door for weird situations where this might not be the case, so if we can scope it properly (per `rowPropertyId` and per `searchContainerId`) maybe that would be enough to cover this use case.

Reply:
> In any case, I think the expectation makes sense as long as we don't open the door for weird situations where this might not be the case, so if we can scope it properly (per rowPropertyId and per searchContainerId) maybe that would be enough to cover this use case.

That means to have an `id` whose format could be something like: `searchContainer_rowPropertyId` as long as `rowPropertyId` is not `null`. Am I right?

Just for simplicity, regarding we are already using `rowId` in our list.jsp, can't we use `searchContainer_rowId`. It's not going to be very useful when `rowId` is just an index but it is not causing any harm and we can make it easier to understand. WDYT?

Reply:
Just as long as we don't use a numeric `id` alone, because that [wouldn't be valid](https://stackoverflow.com/a/1077111/2103996).

Reply:
> That means to have an id whose format could be something like: searchContainer_rowPropertyId as long as rowPropertyId is not null. Am I right?

Not sure we're using the same naming scheme here, but just in case, I'd expect `searchContainerId_rowId` where
- `searchContainerId` is a unique identifier of the search container (which might include the portlet namespace)
- `rowId` is the result of `row.getId()` 
- Generates a [valid](https://www.w3.org/TR/html401/types.html#type-name) HTML `id` value as @wincent wisely pointed out above this. 

And yes, we would only do this if `rowPropertyId != null` since otherwise different elements can have the same identifier on the page regardless of the model.

Reply:
Thanks for the suggestions!

I've sent a new clean pull request in liferay-frontend#986 trying to follow all of them.